nixos/modules/services/networking/networkmanager.nix

```diff
@@ -470,14 +470,25 @@ in {
         mkdir -m 755 -p ${stateDirs}
       '';
 
-      aliases = [ "dbus-org.freedesktop.NetworkManager.service" ];
+      aliases = [
+        "dbus-org.freedesktop.NetworkManager.service"
+        # Add the legacy service name from before #51382 to prevent NetworkManager
+        # from not starting back up:
+        # TODO: remove this once 19.09 is released
```

Comment:
Any reason not to remove this in master as we've forked off 19.09?

Reply:
Hmm, I think we might actually have to remove it during 20.03 branching off.
Still a possibility we could cause an issue for a unstable user?

Reply:
Hmm, right, so that would be for people upgrading from 18.09 -> 20.03 with switch?

> Still a possibility we could cause an issue for a unstable user?

Not entirely sure. But it's there to handle rebuild switch from `legacy-name` to new config with upstream name. So unstable users should have the upstream name already?

Reply:
> Not entirely sure. But it's there to handle rebuild switch from legacy-name to new config with upstream name. So unstable users should have the upstream name already?

Now that I think about it, making guarantees like this for unstable doesn't seem right.
We just want to make sure that rebuild switching from  `19.03 -> 19.09` works fine.
I'm not sure if `18.09 -> 19.03` would be a seamless switch.